testbed/core/tests/test_api.py

```diff
@@ -65,4 +69,283 @@ def test_actor_not_found():
 @pytest.mark.django_db
 def test_outbox_not_found():
     response = APIClient().get(reverse("actor-outbox", kwargs={"pk": 99999}))
-    assert response.status_code == status.HTTP_404_NOT_FOUND
\ No newline at end of file
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+
+
+# LOLA Authentication Tests
+
+"""
+Tests the complete request-response cycle with different authentication states
+to verify that endpoints properly serve enhanced data for LOLA-authenticated requests.
+"""
+class TestLOLAAuthenticationAPI:    
+
+    # Create OAuth application for testing
+    @pytest.fixture
+    def oauth_application(self, db):
+        return Application.objects.create(
+            name="LOLA Test Application",
+            client_type=Application.CLIENT_CONFIDENTIAL,
+            authorization_grant_type=Application.GRANT_AUTHORIZATION_CODE,
+            redirect_uris="http://localhost:8000/callback/",
+        )
+    
+    # Create user for OAuth token testing
+    @pytest.fixture
+    def authenticated_user(self, db):
```

Comment:
Since we're using factories already in this project, some of these strike me very much as Factory objects, and less like fixtures - because they're single objects.  Factory objects can also be customized for an individual test, less easy or desirable with fixtures, so something like AccessTokenFactory can be constructed with the scopes or the expires that you want it to have.

Reply:
You’re right. Most of them could be implemented as factories. I’ll work towards changing the approach.